### PR TITLE
Refactor report fixture

### PR DESCRIPTION
**Pull request by nickpell**
`_report_config_to_fixture` has a lot going on.  This refactor is intended to make it a little bit easier to follow what's happening.

@NoahCarnahan @czue cc: @biyeun

### CODE REVIEW

File: corehq/apps/app_manager/fixtures/mobile_ucr.py [hunk 1]
```diff
@@ -53,16 +53,9 @@ def __call__(self, user, version, last_sync=None, app=None):
...
-    def _report_config_to_fixture(report_config, user):
-        report_elem = E.report(id=report_config.uuid)
-        try:
-            report = get_report_config(report_config.report_id, user.domain)[0]
-        except ResourceNotFound as err:
-            # ReportConfiguration not found
-            raise BadReportConfigurationError('Error getting ReportConfiguration with ID "{}": {}'.format(
-                report_config.report_id, err))
-        data_source = ReportFactory.from_spec(report)
+    @classmethod
```
Comment (czue) [inline corehq/apps/app_manager/fixtures/mobile_ucr.py]: it looks like everything here can still be made to use `@staticmethod`. any reason you changed it?

**Comment by czue** [comment]
looks good. my comment not blocking so feel free to self merge if you don't want to change it.

File: corehq/apps/app_manager/fixtures/mobile_ucr.py [hunk 2]
```diff
@@ -55,14 +55,7 @@ def __call__(self, user, version, last_sync=None, app=None):
...
     def _report_config_to_fixture(report_config, user):
-        report_elem = E.report(id=report_config.uuid)
-        try:
-            report = get_report_config(report_config.report_id, user.domain)[0]
-        except ResourceNotFound as err:
-            # ReportConfiguration not found
-            raise BadReportConfigurationError('Error getting ReportConfiguration with ID "{}": {}'.format(
-                report_config.report_id, err))
-        data_source = ReportFactory.from_spec(report)
+        report, data_source = ReportFixturesProvider._get_report_and_data_source(report_config.report_id, user.domain)
```
Comment (dimagimon) [inline corehq/apps/app_manager/fixtures/mobile_ucr.py]: E501 line too long (118 > 115 characters)

**Comment by nickpell** [comment]
updated

**Pull request merged**